lib/intercom/request.rb

```diff
@@ -72,11 +72,11 @@ def execute(target_base_url = nil, token:, read_timeout: 90, open_timeout: 30, a
             parsed_body
           rescue Intercom::RateLimitExceeded => e
             if @handle_rate_limit
-              seconds_to_retry = (@rate_limit_details[:reset_at] - Time.now.utc).ceil
+              seconds_to_retry = ((@rate_limit_details[:reset_at] || Time.now.utc) - Time.now.utc).ceil
               if (retries -= 1) < 0
                 raise Intercom::RateLimitExceeded, 'Rate limit retries exceeded. Please examine current API Usage.'
               else
-                sleep seconds_to_retry unless seconds_to_retry < 0
+                sleep seconds_to_retry unless seconds_to_retry <= 0
```

Comment:
Perhaps not worth changing this behaviour in this PR to keep the changes in this branch as atomic as possible.
That being said, a more "Ruby" implementation might be
```suggestion
                sleep seconds_to_retry unless seconds_to_retry.negative?
```

Reply:
`.negative?` is (-1, -2, -3), we want to include (0, -1, -2, -3). :o

Reply:
My thinking was that the current `<=` implementation is just there to prevent an ArgumentError when we call `sleep` with a negative argument. Do we really want to protect against a `0` argument here?

My feeling is that changing this condition is not really relevant to the change we are looking to make in this PR so we might be best leaving this condition unchanged.

Reply:
Well if possible would optimise it a bit. I think if you want to sleep something for 0 seconds...its kinda off.

Reply:
If you're keen to make the change then perhaps this?

```suggestion
                sleep seconds_to_retry if seconds_to_retry.positive?
```